axelor-purchase/src/main/java/com/axelor/apps/purchase/service/PurchaseOrderLineServiceImpl.java

```diff
@@ -182,7 +192,39 @@ public BigDecimal getUnitPrice(
       return null;
     }
 
-    BigDecimal price = this.convertUnitPrice(product, taxLine, purchasePrice, purchaseOrder);
+    BigDecimal price =
+        product.getInAti()
+            ? this.convertUnitPrice(product.getInAti(), taxLine, purchasePrice)
+            : purchasePrice;
+
+    return currencyService
+        .getAmountCurrencyConvertedAtDate(
+            purchaseCurrency, purchaseOrder.getCurrency(), price, purchaseOrder.getOrderDate())
+        .setScale(appBaseService.getNbDecimalDigitForUnitPrice(), RoundingMode.HALF_UP);
+  }
+
+  @Override
+  public BigDecimal getInTaxUnitPrice(
+      PurchaseOrder purchaseOrder, PurchaseOrderLine purchaseOrderLine, TaxLine taxLine)
+      throws AxelorException {
+
+    BigDecimal purchasePrice;
+    Currency purchaseCurrency;
+    Product product = purchaseOrderLine.getProduct();
+    SupplierCatalog supplierCatalog =
+        getSupplierCatalog(product, purchaseOrder.getSupplierPartner());
+
+    if (supplierCatalog != null) {
+      purchasePrice = supplierCatalog.getPrice();
+      purchaseCurrency = supplierCatalog.getSupplierPartner().getCurrency();
+    } else {
+      return null;
```

Comment:
Please add javadoc to explain that you return null if supplierCatalog is null, maybe it would be safer to return `BigDecimal.ZERO` instead of null ?